Mage/src/main/java/mage/abilities/effects/common/ExileThenReturnTargetEffect.java

```diff
@@ -40,13 +44,19 @@ protected ExileThenReturnTargetEffect(final ExileThenReturnTargetEffect effect)
         this.putCards = effect.putCards;
         this.yourControl = effect.yourControl;
         this.textThatCard = effect.textThatCard;
+        this.afterEffect = effect.afterEffect;
     }
 
     @Override
     public ExileThenReturnTargetEffect copy() {
         return new ExileThenReturnTargetEffect(this);
     }
 
+    public ExileThenReturnTargetEffect withAfterEffect(Effect afterEffect) {
```

Comment:
either constrain the afterEffect to be a OneShotEffect, or add the code to handle continuous effects